packages/checkbox/theme/lumo/vaadin-checkbox-styles.js

```diff
@@ -119,6 +135,11 @@ registerStyles(
       background-color: var(--lumo-contrast-30pct);
     }
 
+    :host([readonly][checked]) [part='checkbox'],
+    :host([readonly][indeterminate]) [part='checkbox'] {
+      background-color: var(--lumo-contrast-50pct);
```

Comment:
Checked box bg should use new property `--vaadin-checkbox-readonly-checked-background` defaulting to `--lumo-contrast-70pct` (was probably wrong in my proto)

Reply:
Done.